js/ui/control/geolocate_control.js

```diff
@@ -69,12 +73,27 @@ class GeolocateControl extends Evented {
     }
 
     _onSuccess(position) {
-        this._map.jumpTo({
-            center: [position.coords.longitude, position.coords.latitude],
-            zoom: 17,
-            bearing: 0,
-            pitch: 0
-        });
+        if (position.coords.accuracy) {
+            const center = point([position.coords.longitude, position.coords.latitude]);
+            const radius = position.coords.accuracy;
+
+            const right = destination(center, radius, 90, 'meters');
+            const top = destination(center, radius, 0, 'meters');
+            const left = destination(center, radius, -90, 'meters');
+            const bottom = destination(center, radius, 180, 'meters');
+
+            this._map.fitBounds(new LngLatBounds(
+                new LngLat(left.geometry.coordinates[0], bottom.geometry.coordinates[1]),
+                new LngLat(right.geometry.coordinates[0], top.geometry.coordinates[1])
+            ));
```

Comment:
> My understanding of this new functionality is that we set the map's bounds to fit a circle with the radius of the GPS's accuracy. Is that correct?

Exactly. getCurrentPosition can [return an accuracy radius](https://developer.mozilla.org/en-US/docs/Web/API/Coordinates/accuracy). The motivation for this PR is often this location isn't coming from a GPS device so the location might only be city level, so the map should be zoomed out to cover the city, or whatever the accuracy is.

> Is this the behaviour we want in the case where the accuracy is 1 meter?

I think the logic makes sense when the accuracy is say 1km, but not for 1m. I'm thinking of just setting zoom 18 as the upper limit. How does this sound?

> You could borrow the logic from Leaflet to get rid of the Turf dependencies: https://github.com/Leaflet/Leaflet/blob/master/src/geo/LatLng.js#L79-L86

Perfect.